patches/server/0975-Add-event-for-player-editing-sign.patch

```diff
@@ -0,0 +1,32 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: by77er <by77er@gmail.com>
+Date: Mon, 12 Jun 2023 12:56:46 -0400
+Subject: [PATCH] Add event for player editing sign
+
+
+diff --git a/src/main/java/net/minecraft/world/level/block/SignBlock.java b/src/main/java/net/minecraft/world/level/block/SignBlock.java
+index 5ab7df0042391cb621ed78a187dc20333e344186..eab9f846f76b9c6f494880ea6fe4c91c4fa6fa75 100644
+--- a/src/main/java/net/minecraft/world/level/block/SignBlock.java
++++ b/src/main/java/net/minecraft/world/level/block/SignBlock.java
+@@ -144,8 +144,19 @@ public abstract class SignBlock extends BaseEntityBlock implements SimpleWaterlo
+     }
+ 
+     public void openTextEdit(Player player, SignBlockEntity blockEntity, boolean front) {
+-        blockEntity.setAllowedPlayerEditor(player.getUUID());
+-        player.openTextEdit(blockEntity, front);
++        // Paper start
++        org.bukkit.entity.Player bukkitPlayer = (org.bukkit.entity.Player) player.getBukkitEntity();
++        org.bukkit.block.Block bukkitBlock = org.bukkit.craftbukkit.block.CraftBlock.at(blockEntity.getLevel(), blockEntity.getBlockPos());
++        org.bukkit.craftbukkit.block.CraftSign<?> bukkitSign = (org.bukkit.craftbukkit.block.CraftSign<?>) org.bukkit.craftbukkit.block.CraftBlockStates.getBlockState(bukkitBlock);
++        io.papermc.paper.event.player.PlayerOpenSignEvent event = new io.papermc.paper.event.player.PlayerOpenSignEvent(
++            bukkitPlayer,
++            bukkitSign,
++            front ? org.bukkit.block.sign.Side.FRONT : org.bukkit.block.sign.Side.BACK);
++        if (event.callEvent()) {
```

Comment:
Could technically reduce the diff here by changing this to a 
```java
if(!event.callEvent()) return
```

Reply:
True! Done.